spec/classes/agent_upgrade_prepare_spec.rb

```diff
@@ -0,0 +1,113 @@
+require 'spec_helper'
+
+describe 'agent_upgrade::prepare' do
+  context 'on Linux' do
+    let(:facts) {{
+      :osfamily => 'RedHat',
+      :puppet_ssldir   => '/dev/null/ssl',
+      :puppet_config   => '/dev/null/puppet.conf',
+      :mcollective_configured => false,
+    }}
+
+    [true, false].each do |mcollective_configured|
+      context "with mcollective_configured = #{mcollective_configured}" do
+        let(:facts) {{
+          :osfamily => 'RedHat',
+          :puppet_ssldir   => '/dev/null/ssl',
+          :puppet_config   => '/dev/null/puppet.conf',
+          :mcollective_configured => mcollective_configured,
+        }}
+
+        if mcollective_configured
+          it { is_expected.to contain_file('/etc/puppetlabs/mcollective').with_ensure('directory') }
+          it { is_expected.to contain_file('/etc/puppetlabs/mcollective/server.cfg').with({
+            'ensure' => 'file',
+            'source' => '/etc/mcollective/server.cfg',
+          }) }
+        else
+          it { is_expected.to_not contain_file('/etc/puppetlabs/mcollective') }
+        end
+      end
+    end
+
+    ['/etc/puppetlabs', '/etc/puppetlabs/puppet'].each do |dir|
+      it { is_expected.to contain_file(dir).with_ensure('directory') }
+    end
+
+    it { is_expected.to contain_file('/etc/puppetlabs/puppet/puppet.conf').with({
+      'ensure' => 'file',
+      'source' => '/dev/null/puppet.conf',
+    }) }
+
+    it { is_expected.to contain_file('/etc/puppetlabs/puppet/ssl').with({
+      'ensure'  => 'directory',
+      'source'  => '/dev/null/ssl',
+      'backup'  => 'false',
+      'recurse' => 'true',
+    }) }
+
+    ['agent', 'main', 'master'].each do |section|
+      ['allow_variables_with_dashes',
```

Comment:
The main use is to verify that we maintain full code coverage for other changes.